src/SourceBuild/content/test/tests.proj

```diff
@@ -7,4 +7,49 @@
                       Condition="'$(ShortStack)' != 'true' and '$(PortableBuild)' == 'true' and '$(PgoInstrument)' != 'true'" />
   </ItemGroup>
 
-</Project>
\ No newline at end of file
+  <Target Name="RunCoreTests">
+    <MSBuild Projects="$(MSBuildThisFileFullPath)"
+             Properties="Configuration=$(Configuration)"
+             ContinueOnError="true"
+             Targets="Build;VSTest" />
+
+    <PropertyGroup>
+      <CoreTestsResult>$(MSBuildLastTaskResult)</CoreTestsResult>
+    </PropertyGroup>
+  </Target>
```

Comment:
We shouldn't need a separate target for this. MSBuild can invoke multiple entry point targets: `VSTest;RunScenarioTests`.

Reply:
The problem with this is that any failure in the `VSTest` target will stop execution, preventing the `RunScenarioTests` target from running. So you don't get a full view of all test results until you can get the first target to be successful.

Reply:
> This guarantees that when I do `dotnet test test/tests.proj`, we use the same same entry point as when doing `build.cmd -test`.

Why is attempting to support `dotnet test` even a scenario? There's a whole bunch of MSBuild properties that need to be set in order for the tests to do the right thing. This is all done by the build script.

Reply:
We use the Traversal msbuild sdk which is quite flexible. Does one of these setting help? https://github.com/microsoft/MSBuildSdks/blob/32403937dc2bb90f74c2156b9e140538b3f25d51/src/Traversal/Sdk/Traversal.targets#L210-L211

Reply:
I quickly tried this out:

```xml
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <OutputType>Exe</OutputType>
    <TargetFramework>net9.0</TargetFramework>
    <ImplicitUsings>enable</ImplicitUsings>
    <Nullable>enable</Nullable>
  </PropertyGroup>

  <Target Name="A">
    <Error Text="A target" ContinueOnError="WarnAndContinue" />
  </Target>

  <Target Name="B">
    <Error Text="B target" />
  </Target>

</Project>
```

> PS C:\temp\multitarg> dotnet msbuild -t:"A;B"
MSBuild version 17.10.0-preview-24101-01+07fd5d51f for .NET
  multitarg succeeded with warnings (0.0s)
    C:\temp\multitarg\multitarg.csproj(11,5): warning : A target [C:\temp\multitarg\multitarg.csproj]
  multitarg failed with errors (0.0s)
    C:\temp\multitarg\multitarg.csproj(15,5): error : B target [C:\temp\multitarg\multitarg.csproj]

---

That said, I think it would make more sense to just define the Test/VSTest target in scenario-tests.proj and just let Traversal SDK invoke that target as it already does with the other two test projects. In my head tests.proj could just look like this:

```xml
<Project Sdk="Microsoft.Build.Traversal">

  <ItemGroup>
    <ProjectReference Include="Microsoft.DotNet.SourceBuild.SmokeTests\Microsoft.DotNet.SourceBuild.SmokeTests.csproj"
                      Condition="'$(DotNetBuildSourceOnly)' == 'true'" />
    <ProjectReference Include="Microsoft.DotNet.UnifiedBuild.Tests\Microsoft.DotNet.UnifiedBuild.Tests.csproj"
                      Condition="'$(ShortStack)' != 'true' and '$(PortableBuild)' == 'true' and '$(PgoInstrument)' != 'true'" />
    <ProjectReference Include="scenario-tests.proj" />
  </ItemGroup>

</Project>
```

This would have the benefit of keeping a single entry point target and reduce complexity from tests.proj and move that into scenario-tests.proj itself.

Reply:
> I think it would make more sense to just define the VSTest target in scenario-tests.proj and just let Traversal SDK invoke that target as it already does with the other two test projects.

Unfortunately, that approach doesn't work. I can't override the `VSTest` target. It ends up getting overridden because `Microsoft.TestPlatform.ImportAfter.targets` and `Microsoft.TestPlatform.targets` get imported after the definition of scenario-tests.proj.

Reply:
> ```xml
>   <Target Name="A">
>     <Error Text="A target" ContinueOnError="WarnAndContinue" />
>   </Target>
> 
>   <Target Name="B">
>     <Error Text="B target" />
>   </Target>
> ```

This isn't really possible without introducing a new target. I can't influence the `VSTest` target to have the behavior that you've shown in target `A`. This is what my original implementation solves.

Reply:
Fixed in https://github.com/dotnet/installer/pull/19222/commits/1c9cdd7248d20ace14d028008d1edf7bb395ba0f